File: webapp/src/modules/v2/pocketbase/pocketbase.ts
// SPDX-FileCopyrightText: 2025 Forkbomb BV
//
// SPDX-License-Identifier: AGPL-3.0-or-later

import type * as pb from 'pocketbase';
import type { Simplify } from 'type-fest';

import GenericPocketBase from 'pocketbase';

import { pb as defaultPocketbaseClient } from '@/pocketbase';

//

export type Pocketbase = GenericPocketBase;

export type QueryOptions = Simplify<
	pb.FullListOptions | pb.RecordListOptions | pb.RecordOptions | pb.ListOptions
>;

export const defaultClient = defaultPocketbaseClient;

File: pkg/workflowengine/activities/pocketbase.go
// SPDX-FileCopyrightText: 2025 Forkbomb BV
//
// SPDX-License-Identifier: AGPL-3.0-or-later

package activities

import (
	"errors"
	"sync"

	"github.com/pocketbase/pocketbase/core"
)

var pocketBaseApp core.App
var pocketBaseAppMu sync.RWMutex

func SetPocketBaseApp(app core.App) {
	pocketBaseAppMu.Lock()
	defer pocketBaseAppMu.Unlock()
	pocketBaseApp = app
}

func getPocketBaseApp() (core.App, error) {
	pocketBaseAppMu.RLock()
	defer pocketBaseAppMu.RUnlock()
	if pocketBaseApp == nil {
		return nil, errors.New("pocketbase app not configured")
	}
	return pocketBaseApp, nil
}
